File: 0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits/0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits.java
import java.util.*;

class Solution {
    public int maximumSum(int[] nums) {
        int ans = -1;
        Map<Integer, Integer> mp = new HashMap<>();

        for (int x : nums) {
            int sumDigits = 0, tmp = x;
            while (tmp > 0) {
                sumDigits += tmp % 10;
                tmp /= 10;
            }

            if (mp.containsKey(sumDigits)) {
                ans = Math.max(ans, mp.get(sumDigits) + x);
                mp.put(sumDigits, Math.max(mp.get(sumDigits), x));
            } else {
                mp.put(sumDigits, x);
            }
        }
        
        return ans;
    }
}

File: 0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits/0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits.js
var maximumSum = function(nums) {
    let ans = -1;
    let mp = new Map();

    for (let x of nums) {
        let sumDigits = [...String(x)].reduce((a, b) => a + Number(b), 0);
        if (mp.has(sumDigits)) {
            ans = Math.max(ans, mp.get(sumDigits) + x);
            mp.set(sumDigits, Math.max(mp.get(sumDigits), x));
        } else {
            mp.set(sumDigits, x);
        }
    }
    
    return ans;
};
